### PR TITLE
Add ability to subscribe, unsubscribe and resubscribe to gossip topics

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
Rework how gossip managers are structured so that there are explicit calls to `subscribe` rather than it being done in the constructor.  Apart from being cleaner code it also provides support for unsubscribing then resubscribing to gossip topics.

## Fixed Issue(s)
#4784 

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/AbstractGossipManager.java [hunk 1]
```diff
@@ -57,48 +56,25 @@ protected AbstractGossipManager(
...
   @Override
-  public void shutdown() {
-    if (shutdown.compareAndSet(false, true)) {
-      // Close gossip channels
-      channel.close();
-      publisher.unsubscribe(subscriberId);
+  public void unsubscribe() {
+    if (channel.isPresent()) {
+      channel.get().close();
+      channel = Optional.empty();
```
Comment (rolfyone) [inline networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/AbstractGossipManager.java:77]: could lead to concurrency fun with the subscribe, do they need to be synchronized?

Comment (ajsutton) [reply networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/AbstractGossipManager.java:77]: They're synchronized back up at the `GossipForkManager` level.

Comment (ajsutton) [reply networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/AbstractGossipManager.java:77]: hmm, except potentially for the ones that use a gossip publisher... Will go check those.

**Review by rolfyone — APPROVED** [review]
LGTM i was hoping that subscribe/unsubscribe might be enough, but its definitely safer to include the publish

**Pull request merged**